mne_bids/tests/test_write.py

```diff
@@ -1469,3 +1470,142 @@ def test_write_does_not_alter_events_inplace():
+                           datatype, overwrite):
+    """Test marking channels of an existing BIDS dataset as "bad"."""
+
+    # Setup: Create a fresh BIDS dataset.
+    bids_root = _TempDir()
+    bids_path = _bids_path.copy().update(root=bids_root, datatype='meg',
+                                         suffix='meg')
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+    event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
+                'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
+    events_fname = op.join(data_path, 'MEG', 'sample',
+                           'sample_audvis_trunc_raw-eve.fif')
+    raw = _read_raw_fif(raw_fname, verbose=False)
+    raw.info['bads'] = []
+    write_raw_bids(raw, bids_path=bids_path, events_data=events_fname,
+                   event_id=event_id, verbose=False)
+
+    channels_fname = _find_matching_sidecar(bids_path, suffix='channels',
+                                            extension='.tsv')
+
+    if drop_status_col:
+        # Remove `status` column from the sidecare TSV file.
+        tsv_data = _from_tsv(channels_fname)
+        del tsv_data['status']
+        _to_tsv(tsv_data, channels_fname)
+
+    if drop_description_col:
+        # Remove `status_description` column from the sidecare TSV file.
+        tsv_data = _from_tsv(channels_fname)
+        del tsv_data['status_description']
+        _to_tsv(tsv_data, channels_fname)
+
+    # Test that we raise if number of channels doesn't match number of
+    # descriptions.
+    if (descriptions is not None and
+            len(_ensure_list(ch_names)) != len(_ensure_list(descriptions))):
+        with pytest.raises(ValueError, match='must match'):
+            mark_bad_channels(ch_names=ch_names, descriptions=descriptions,
+                              bids_path=bids_path, overwrite=overwrite)
+        return
+
+    # Test that we raise if we encounter an unknown channel name.
+    if any([ch_name not in raw.ch_names
+            for ch_name in _ensure_list(ch_names)]):
+        with pytest.raises(ValueError, match='not found in dataset'):
+            mark_bad_channels(ch_names=ch_names, descriptions=descriptions,
+                              bids_path=bids_path, overwrite=overwrite)
+        return
+
+    if not overwrite:
+        # Mark `existing_ch_names` as bad in raw and sidecar TSV before we
+        # begin our actual tests, which should then add additional channels
+        # to the list of bads, retaining the ones we're specifying here.
+        mark_bad_channels(ch_names=existing_ch_names,
+                          descriptions=existing_descriptions,
+                          bids_path=bids_path, overwrite=True)
+        _bids_validate(bids_root)
+        raw = read_raw_bids(bids_path=bids_path, verbose=False)
+        # XXX Order is not preserved
```

Comment:
```suggestion
        # Order is not preserved
```
it's not XXX as we don't expect a fix right?